src/field/trainer.ts

```diff
@@ -425,6 +425,12 @@ export default class Trainer extends Phaser.GameObjects.Container {
       }
     }
 
+    // Prompts reroll of party member species if species already present in the enemy party
+    if (this.checkDuplicateSpecies(ret)) {
+      console.log("Duplicate species detected, prompting reroll...");
+      retry = true;
+    }
+
     if (retry && (attempt || 0) < 10) {
       console.log("Rerolling party member...");
       ret = this.genNewPartyMemberSpecies(level, strength, (attempt || 0) + 1);
```

Comment:
```suggestion
    if (retry && (attempt ?? 0) < 10) {
      console.log("Rerolling party member...");
      ret = this.genNewPartyMemberSpecies(level, strength, (attempt ?? 0) + 1);
```
Might as well fix nearby.